scala/wdl_scala/src/main/scala/wdl_scala/RuntimeAttributes.scala

```diff
@@ -0,0 +1,76 @@
+package wdl_scala
+
+import wdl_scala.AstTools.{AstNodeName, EnhancedAstNode}
+import wdl_scala.expression.NoFunctions
+import wdl_scala.values._
+import wdl_scala.parser.WdlParser.{Ast, AstList}
+import wdl_scala.parser.MemorySize
+
+import scala.collection.JavaConverters._
+import scala.language.postfixOps
+import scalaz.Scalaz._
+import scalaz._
+
+case class RuntimeAttributes(attrs: Map[String, String])
+
+object RuntimeAttributes {
+  def apply(ast: Ast): RuntimeAttributes = {
+    val asts = ast.findAsts(AstNodeName.Runtime)
+    if (asts.size > 1) throw new UnsupportedOperationException("Only one runtime block may be defined per task")
+    val astList = asts.headOption map { _.getAttribute("map").asInstanceOf[AstList] }
+    val attrMap = astList map processRuntimeAttributes getOrElse Map.empty[String, String]
+    attrMap.get("memory") foreach { validateMemory }
+
+    RuntimeAttributes(attrMap)
+  }
+
+  /**
+    * Checks that a provided memory value is valid, if not throws an IllegalArgumentException
+    */
+  private def validateMemory(value: String): Unit = {
+    validateMemStringInGb (value) match {
+      case Failure(f) => throw new IllegalArgumentException("Invalid memory value specified:\n" + f.toList.mkString("\n"))
+      case Success(_) => ()
+    }
+  }
+
+  private def validateMemStringInGb(mem: String): ValidationNel[String, Double] = {
```

Comment:
looks like `ErrorOr` should find its way over to this repo, maybe even applied to a feline library